### PR TITLE
DESENG-55. File uploader modal saves, updates, and returns files

### PR DESCRIPTION
**Pull request by Baelx**
[] Wrote a test or tests for code you added.
[x] Linted the files you touched.
[x] Wrote a doc block for each method you added.
[x] Add documentation to the README

PARTIAL PR.
- Trying to split this feature up into multiple PRs. One more should do it to finish off this ticket.

### CODE REVIEW

File: src/app/file-upload-modal/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ The modal technology used with is [NGX Smart modal](https://github.com/maximelaf
 ```
 this.ngxSmartModalService.setModalData({
   ...yourOptions
-}, 'file-upload-modal');
+}, 'file-upload-modal', true);
```
Comment (ShawnTurple) [inline src/app/file-upload-modal/README.md:18]: What is the true parameter used for?

File: src/app/file-upload-modal/file-upload-modal.component.html [hunk 2]
```diff
@@ -1,62 +1,75 @@
...
+  <div class="spinner-container" *ngIf="loading">
+    <div class="spinner-new rotating"></div>
+  </div>
+  <ul *ngIf="fileList.length > 0">
+    <li
+      tabindex="0"
+      class="file"
+      *ngFor="let file of fileList; index as i"
+      (click)=selectFile(file)
+      (keypress)="keyPress($event, file)"
```
Comment (ShawnTurple) [inline src/app/file-upload-modal/file-upload-modal.component.html:37]: just wondering why you add $ to event variable, is this an Angular thing..

File: src/app/file-upload-modal/file-upload-modal.component.html [hunk 3]
```diff
@@ -1,62 +1,75 @@
...
     <form #fileInfoForm [ngClass]="{ 'no-file-selected': 0 === selectedFiles.length }">
       <p *ngIf="0 === selectedFiles.length" class="no-file-selected">
         <mat-icon aria-hidden="false" [inline]="true">description</mat-icon>
         No file selected
       </p>
-      <div *ngIf="selectedFiles.length > 0">
+      <div *ngIf="selectedFiles.length > 0" class="form-content">
         <label>File name</label>
-        <p>{{ selectedFiles[0].name }}</p>
+        <p>{{ selectedFiles[0].documentFileName }}</p>
```
Comment (ShawnTurple) [inline src/app/file-upload-modal/file-upload-modal.component.html:56]: i'm a bit confused, what if selectedFiles.length > 1, how do multiples selected files get shown.

File: src/app/file-upload-modal/file-upload-modal.component.scss [hunk 4]
```diff
@@ -1,5 +1,36 @@
...
+    @content;
+  }
+}
+
+.spinner-container {
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  background: #fff;
```
Comment (ShawnTurple) [inline src/app/file-upload-modal/file-upload-modal.component.scss:26]: I suspect this should be using a sass variable if we are using sass.

File: src/app/file-upload-modal/file-upload-modal.component.ts [hunk 5]
```diff
@@ -67,22 +94,70 @@ export class FileUploadModalComponent implements OnInit {
...
+   * Then puts the new files at the beginning of the fileList.
+   *
+   * @param queuedFiles The files queued for upload.
+   * @returns {void}
+   */
+  queueFilesForUpload( queuedFiles: FileList ): void {
+    Array.from(queuedFiles).forEach((file: File) => {
+      let documentForm = new DocumentForm();
+      // Create a temporary pseudo-random string ID to help with sorting.
+      documentForm._id = Math.random().toString(36).slice(2);
```
Comment (ShawnTurple) [inline src/app/file-upload-modal/file-upload-modal.component.ts:151]: So I'm assuming this Math.random() is only temporary and not used for uniqueness across all images being uploaded at any given time, but just for this particular instance.

**Review by ShawnTurple — APPROVED** [review]
Some minor questions and perhaps you can change the #fff to $white.

File: src/app/file-upload-modal/README.md (hunk 1 above)
Comment (Baelx) [reply src/app/file-upload-modal/README.md:18]: The 3rd parameter of setModalData is to overwrite old data attached to a particular modal: https://maximelafarie.com/ngx-smart-modal/#/api

We need to do this so we don't have options from an older invocation of the modal sticking around.

File: src/app/file-upload-modal/file-upload-modal.component.html (hunk 2 above)
Comment (Baelx) [reply src/app/file-upload-modal/file-upload-modal.component.html:37]: I thought this was required with Angular but your comment got me to look into it. AngularJS(an older version of Angular) used to prefix its internal variables with a dollar sign. Event handlers had to use this convention too. These days it doesn't matter. 

I'll no longer use the dollar sign going forward.

File: src/app/file-upload-modal/file-upload-modal.component.scss (hunk 4 above)
Comment (Baelx) [reply src/app/file-upload-modal/file-upload-modal.component.scss:26]: Missed this! Thanks for the catch

File: src/app/file-upload-modal/file-upload-modal.component.ts (hunk 5 above)
Comment (Baelx) [reply src/app/file-upload-modal/file-upload-modal.component.ts:151]: You're right in that it is temporary. We basically need a random ID for every file that is uploaded at once. We expect that to be around 2 or 3 files so I feel this should be random enough.

File: src/app/file-upload-modal/file-upload-modal.component.html (hunk 3 above)
Comment (Baelx) [reply src/app/file-upload-modal/file-upload-modal.component.html:56]: The file upload modal uses a dynamic form which focuses on the 1st of the selected files in the array. This is similar to how the Wordpress media library allows you to select multiple files at once. The most recently-clicked item will fill its details out in the panel on the right and therein you can update its alt tag.

**Comment by Baelx** [comment]
Thanks @ShawnTurple . I will update the non-variable colour I missed in the last PR for this.

**Pull request merged**